api/storage_service.cc

```diff
@@ -668,23 +668,16 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         auto keyspace = validate_keyspace(ctx, req->param);
         auto table_infos = parse_table_infos(keyspace, ctx, req->query_parameters, "cf");
         apilog.debug("force_keyspace_compaction: keyspace={} tables={}", keyspace, table_infos);
+
+        auto& cm = db.local().get_compaction_manager();
+        auto compaction_module_ptr = cm.get_compaction_module().shared_from_this();
+        auto task_impl_ptr = std::make_unique<major_keyspace_compaction_task_impl>(compaction_module_ptr, std::move(keyspace), db, table_infos);
+        auto task = co_await compaction_module_ptr->make_task(std::move(task_impl_ptr), {});
+        task->start();
```

Comment:
I've seen this boilerplate many times now (in other PRs too). Would it be possible to create a factory method which does all this? Something like:

```c++
template <typename TaskType, typename... TaskArgs>
future<task_type> create_and_start_task(module_type& module, TaskArgs&&... args) {
         auto task_impl_ptr = std::make_unique<TaskType>(module.shared_from_this(), std::forward<TaskArgs>(args)...);
        auto task = co_await module.make_task(std::move(task_impl_ptr), {});
        task->start();
        co_return task;
}
````